PurchasesCoreSwift/Public/EntitlementInfos.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  EntitlementInfos.swift
+//  PurchasesCoreSwift
+//
+//  Created by Joshua Liebowitz on 6/28/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+/**
+ This class contains all the entitlements associated to the user.
+ */
+@objc(RCEntitlementInfos) public class EntitlementInfos: NSObject {
+    /**
+     Dictionary of all EntitlementInfo (`RCEntitlementInfo`) objects (active and inactive) keyed by entitlement
+     identifier. This dictionary can also be accessed by using an index subscript on EntitlementInfos, e.g.
+     `entitlementInfos[@"pro_entitlement_id"]`.
+     */
+    @objc public let all: [String: EntitlementInfo]
+
+    /**
+     Dictionary of active EntitlementInfo (`RCEntitlementInfo`) objects keyed by entitlement identifier.
+     */
+    @objc public var active: [String: EntitlementInfo] {
+        return self.all.filter { $0.value.isActive }
+    }
+
+    @objc public init(entitlementsData: [String: Any]?,
+                      purchasesData: [String: Any],
+                      dateFormatter: DateFormatter,
+                      requestDate: Date?) {
+        guard let entitlementsData = entitlementsData else {
+            self.all = [:]
+            return
+        }
+
+        var entitlementInfos: [String: EntitlementInfo] = [:]
+        entitlementsData.forEach { identifier, entitlement in
+            guard let entitlement = entitlement as? [String: Any] else {
+                return
+            }
+
+            guard let productIdentifier = entitlement["product_identifier"] as? String else {
+                return
+            }
+
+            let productData = purchasesData[productIdentifier]
+            guard let productData = productData as? [String: Any] else {
```

Comment:
perhaps these could be grouped into a single clause? I don't have a strong preference, though, readability for this is fine (and miles better than what it used to be before the migration)

Reply:
To me, this reads better than grouping them all.